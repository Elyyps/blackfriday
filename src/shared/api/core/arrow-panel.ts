import { Link } from "./link";

export class ArrowPanel {
  public brands?: string[];
  public icon?: string;
  public text: Link;
}
