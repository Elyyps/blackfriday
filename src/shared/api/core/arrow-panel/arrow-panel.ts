export class ArrowPanel {
  public brands?: string[];
  public icon?: string;
  public text: string;
  public url: string;
}
