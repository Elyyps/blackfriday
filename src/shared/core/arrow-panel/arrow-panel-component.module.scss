.arrow-panel {
  display: flex;
  align-items: center;
  position: relative;
  background: $global-background;
  border-radius: $global-border-radius;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.08);
  padding: 16px 42px 16px 16px;
  height: 52px;
  margin: 8px;
  overflow: hidden;
  min-height: 52px;
  cursor: pointer;
  box-sizing: border-box;
  svg {
    position: absolute;
    right: 16px;
    top: 50%;
    transform: translateY(-50%);
    height: 20px;
    path {
      stroke: #1c1c1c;
    }
  }

  &:hover {
    text-decoration: none;
    color: currentColor;
  }
  @media (max-width: $breakpoint-small-max) {
    margin: 0;
    height: 40px;
    border-bottom: #e0e0e0 1px solid;
    width: 100%;
  }
  &__title {
    color: $global-color;
    font-weight: bold;
    font-family: $global-font-family-secondary;
    line-height: 1.2;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;

    @media (max-width: $breakpoint-xsmall-max) {
      font-size: 14px;
    }
  }

  &__body {
    display: flex;
    align-items: center;
  }
  &__list-items {
    color: $gray-light;
    font-weight: 600;
    font-family: $global-font-family-secondary;
    line-height: 1.2;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;
    font-size: 11px;
  }

  &__image {
    margin-right: 16px;
    width: 40px;
    flex-shrink: 0;
    text-align: center;

    svg {
      width: 100%;
      height: 100%;
    }
  }

  &--small {
    padding-top: 8px;
    padding-bottom: 8px;
  }

  &--list-items {
    justify-content: space-between;
  }
}
