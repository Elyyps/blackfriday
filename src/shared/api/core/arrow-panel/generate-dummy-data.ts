import { IArrowPanelProps } from "@app/core/arrow-panel";
import party from "@assets/icons/party.svg";
import iears from "@assets/dummy-images/product-thumbnails/iears.png";
import imac from "@assets/dummy-images/product-thumbnails/imac.png";
import iphone from "@assets/dummy-images/product-thumbnails/iphone.png";
import loader from "@assets/dummy-images/product-thumbnails/loader.png";
import macbook from "@assets/dummy-images/product-thumbnails/macbook.png";
import mouse from "@assets/dummy-images/product-thumbnails/mouse.png";

export const generateArrowPanelData = (): IArrowPanelProps[] => {
  const data: IArrowPanelProps[] = [
    {
      image: iears,
      link: { url: "/", title: "Apple Airpods" }
    },
    {
      image: imac,
      link: { url: "/", title: "Apple iPhones" }
    },
    {
      image: iphone,
      link: { url: "/", title: "Apple MacBooks" }
    },
    {
      image: loader,
      link: { url: "/", title: "Apple iMacs" }
    },
    {
      image: macbook,
      link: { url: "/", title: "Categorie item met een lange naam" }
    },
    {
      image: mouse,
      link: { url: "/", title: "Categorie item met een nog langere naam" }
    },
    {
      link: { url: "/", title: "Categorie item zonder afbeelding" }
    },
    {
      icon: party,
      link: { url: "/", title: "Categorie met icoon" }
    }
  ];

  return data;
};
