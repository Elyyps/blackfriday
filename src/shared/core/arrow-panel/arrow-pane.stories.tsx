import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { ArrowPanelComponent } from "./arrow-panel.component";
import CardIcon from "@assets/icons/store.svg";
import { withProvider } from "@app/util";

storiesOf("Arrow-panel", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <ArrowPanelComponent title="Productdeals" icon={CardIcon} link={"/"} />)
  .add("Emtpy implementation", () => (
    <ArrowPanelComponent title="Productdeals" brands={["Nike", "Adidas", "New Balance", "Versace"]} link={"/"} />
  ));
