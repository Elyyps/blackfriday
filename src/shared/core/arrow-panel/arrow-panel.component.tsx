import * as React from "react";
import styles from "./arrow-panel-component.module.scss";
import { IconComponent, ImageComponent } from "@app/core";
import { Link as ILink } from "@app/api/core/link";
import { CHEVRON_RIGHT } from "@app/constants/icons";
import { Link } from "react-router-dom";

export interface IArrowPanelProps {
  icon?: string;
  image?: string;
  link: ILink;
}

const ArrowPanel = ({ link: { title, url }, icon, image }: IArrowPanelProps) => (
  <Link to={url} className={`${styles["card-simple"]}`}>
    <div className={styles["content"]}>
      {(image || icon) && (
        <div className={styles["image-container"]}>
          {image && (
            <ImageComponent alt="Arrow icon" height={"100%"} width={"100%"} className={styles["image"]} src={image} />
          )}
          {icon && <ImageComponent alt="Arrow icon" height={"60%"} width={"60%"} src={icon} />}
        </div>
      )}
      <div className={styles["title"]}>{title}</div>
    </div>
    <div className={styles["icon"]}>
      <IconComponent icon={CHEVRON_RIGHT} size="10px" strokeColor="#474747" strokeWidth="3px" />
    </div>
  </Link>
);
export { ArrowPanel };
