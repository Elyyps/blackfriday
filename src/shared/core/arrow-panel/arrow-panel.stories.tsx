import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { ArrowPanel } from "./arrow-panel.component";
import CardIcon from "@assets/icons/store.svg";
import { withProvider } from "@app/util";

storiesOf("Arrow-panel", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <ArrowPanel icon={CardIcon} link={{ title: "Productdeals", url: "/" }} />);
