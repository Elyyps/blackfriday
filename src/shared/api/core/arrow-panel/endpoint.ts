import { generateArrowPanelData } from "./generate-dummy-data";

export const generateTabsDummyData = generateArrowPanelData();
