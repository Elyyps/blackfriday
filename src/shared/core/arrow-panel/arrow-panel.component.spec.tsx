import React from "react";
import { shallow } from "enzyme";

import CardIcon from "@assets/icons/store.svg";
import { ArrowPanelComponent } from "./arrow-panel.component";

describe("[arrow-panel]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<ArrowPanelComponent title="Productdeals" icon={CardIcon} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
