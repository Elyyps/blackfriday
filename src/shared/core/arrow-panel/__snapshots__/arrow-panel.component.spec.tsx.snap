// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[arrow-panel] should render component without crashing 1`] = `
<Link
  className="card-simple"
  to="/"
>
  <div
    className="content"
  >
    <div
      className="image-container"
    >
      <ImageComponent
        alt="Arrow"
        className="image"
        src="store.svg"
      />
    </div>
    <div
      className="title"
    >
      Productdeals
    </div>
  </div>
  <div
    className="icon"
  >
    <IconComponent
      icon={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "chevron-right.svg",
        }
      }
      size="10px"
      strokeColor="#474747"
      strokeWidth="3px"
    />
  </div>
</Link>
`;
